### PR TITLE
Api functionality/sm

### LINKED ISSUE #6: Set up API for all movies

### PR DESCRIPTION
**Pull request by scmistele**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context.
 -Removed sample data file
 -Added Heroku API data 
 -Added two fetch calls, one for all movies and one for single movie
 -Removed unused filepaths from components
 -Refactored the showSingleMovie method
 -Updated JSX in MovieInfo component to show all single movie data

## List any issues this closes
#6 #19 #20 #22 #21 #23

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Refactor
- [ ] Documentation update

## Check the correct box

- [x] This broke nothing
- [ ] This broke some stuff
- [ ] This broke everything

## Checklist:

- [x] I have linked the appropriate Project Board in this PR
- [x] My code follows the style guidelines of this project
- [x] My code has no unused/commented out code
- [x] I have performed a self-review of my code
- [ ] I have fully tested my code

### CODE REVIEW

**Comment by gjax78** [comment]
For this branch, we did driver/navigator paired programming. All work was done and reviewed together.

File: src/App.js [hunk 1]
```diff
@@ -1,27 +1,36 @@
...
+      movieData: [],
       selectedMovie: null
     }
   }
 
+  componentDidMount = () => {
+    apiData.allMovieData()
+    .then(data => this.setState({movieData: data.movies}))
+    .catch((error) => {
+      console.log('Error:', error)
```
Comment (sertmer) [inline src/App.js:24]: Do you have any plans for this error down the road - maybe storing it in state so we can eventually surface it to the user?

File: src/App.js [hunk 2]
```diff
@@ -1,27 +1,36 @@
...
+    })
+  }
+
   showSingleMovie = (id) => {
-    const singleMovie = this.state.movieData.find(movie => {
-      return movie.id === id
+    const singleMovie = apiData.singleMovieData(id)
+    .then(movie => this.setState({selectedMovie: movie.movie}))
+    .catch((error) => {
+      console.log('Error:', error)
```
Comment (sertmer) [inline src/App.js:32]: Same question here as above

File: src/MovieInfo.js [hunk 3]
```diff
@@ -1,22 +1,19 @@
...
-          <p>runtime</p>
-          <p>genre</p>
-          <p>tagline</p>
-          <p>overview</p>
+          <p>{movie.budget}</p>
+          <p>{movie.revenue}</p>
+          <p>{movie.runtime}</p>
+          <p>{movie.genres}</p>
+          <p>{movie.tagline}</p>
+          <p>{movie.overview}</p>
```
Comment (sertmer) [inline src/MovieInfo.js:16]: Good use case for destructuring here - so you don't have to type `movie` over and over 😄

File: src/apiCalls.js [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+      if(!response.ok) {
+        throw new Error('Refresh browser')
+      } else {
+        return response.json()
+      }
+    })
+  }
+}
+
+export default apiData;
```
Comment (sertmer) [inline src/apiCalls.js:24]: These two fetches look relatively similar. The endpoint is the only major difference... maybe we could send the entire URL through as an argument? Then we could have one fetch that can do both!

**Review by sertmer — COMMENTED** [review]
Good stuff overall! Check my comments for some questions and suggestions!

**Pull request merged**